src/main/java/org/opensrp/repository/postgres/ClientMigrationFileRepositoryImpl.java

```diff
@@ -0,0 +1,256 @@
+                convertedMigrationFiles.add(migrationFile);
+            }
+        }
+
+        return convertedMigrationFiles;
+    }
+
+    @Nullable
+    @Override
+    public ClientMigrationFile getClientMigrationFileById(long id) {
+        if (id == 0) {
+            return null;
+        }
+
+        return convert(clientMigrationFileMapper.selectByPrimaryKey(id));
+    }
+
+    @Override
+    public ClientMigrationFile getClientMigrationFileByIdentifier(String identifier) {
+        if (identifier == null) {
+            return null;
+        }
+
+        return convert(clientMigrationFileMapper.selectByIdentifier(identifier));
+    }
+
+    @Override
+    public List<ClientMigrationFile> getAll(int limit) {
+        if (limit == 0) {
+            return new ArrayList<>();
+        }
+
+        ArrayList<ClientMigrationFile> convertedMigrationFiles = new ArrayList<>();
+        List<org.opensrp.domain.postgres.ClientMigrationFile> psqlMigrationFiles = clientMigrationFileMapper.getAll(0, limit);
+
+        for (org.opensrp.domain.postgres.ClientMigrationFile psqlMigrationFile: psqlMigrationFiles) {
+            ClientMigrationFile migrationFile = convert(psqlMigrationFile);
+
+            if (migrationFile != null) {
+                convertedMigrationFiles.add(migrationFile);
+            }
+        }
+
+        return convertedMigrationFiles;
+    }
+
+    private org.opensrp.domain.postgres.ClientMigrationFile convert(@Nullable ClientMigrationFile clientMigrationFile, Long primaryKey) {
+        if (clientMigrationFile == null) {
+            return null;
+        }
+
+        org.opensrp.domain.postgres.ClientMigrationFile psqlClientMigrationFile = new org.opensrp.domain.postgres.ClientMigrationFile();
+
+        if (primaryKey != null) {
+            psqlClientMigrationFile.setId(primaryKey);
+        } else {
+            psqlClientMigrationFile.setId(clientMigrationFile.getId());
+        }
+
+        psqlClientMigrationFile.setIdentifier(clientMigrationFile.getIdentifier());
+        psqlClientMigrationFile.setFilename(clientMigrationFile.getFilename());
+        psqlClientMigrationFile.setOnObjectStorage(clientMigrationFile.getOnObjectStorage());
+        psqlClientMigrationFile.setObjectStoragePath(clientMigrationFile.getObjectStoragePath());
+        psqlClientMigrationFile.setJurisdiction(clientMigrationFile.getJurisdiction());
+        psqlClientMigrationFile.setVersion(clientMigrationFile.getVersion());
+        psqlClientMigrationFile.setManifestId(clientMigrationFile.getManifestId());
+        psqlClientMigrationFile.setFileContents(clientMigrationFile.getFileContents());
+
+        if (clientMigrationFile.getCreatedAt() != null) {
+            psqlClientMigrationFile.setCreatedAt(clientMigrationFile.getCreatedAt());
+        }
+
+        return psqlClientMigrationFile;
+    }
+
+
+    private ClientMigrationFile convert(@Nullable org.opensrp.domain.postgres.ClientMigrationFile clientMigrationFile) {
```

Comment:
This function looks identical to 
`private org.opensrp.domain.postgres.ClientMigrationFile convert(@Nullable ClientMigrationFile clientMigrationFile, Long primaryKey)`

it can be simplified to
```
private org.opensrp.domain.postgres.ClientMigrationFile convert(@Nullable ClientMigrationFile clientMigrationFile){
return org.opensrp.domain.postgres.ClientMigrationFile(clientMigrationFile, null);
{
````

Reply:
Oh! Yea, it seems the two are almost not similar. But they convert from the domains in `domain.postgres` and `domain`. One is a domain in `org.opensrp.domain.postgres` while the other is in `org.opensrp.domain`

Reply:
Noted